Makefile

```diff
@@ -0,0 +1,8 @@
+.PHONY: build
+build:
+	@go build github.com/envoyproxy/gateway/cmd/envoy-gateway
```

Comment:
we can deal with cross compilation and LD flags later but we should at least save the binary in a separate dir such as `bin` or `build`